src/cve/utils/tracing.py

```diff
@@ -0,0 +1,313 @@
+    
+    # Add OTLP exporter for Jaeger/OpenTelemetry collector only if enabled
+    if AGENT_MORPHEUS_JAEGER_ENABLED:
+        logger.info(f"Connecting to tracing collector at {AGENT_MORPHEUS_JAEGER_ENDPOINT}")
+        try:
+            otlp_exporter = OTLPSpanExporter(endpoint=AGENT_MORPHEUS_JAEGER_ENDPOINT, insecure=True)
+            provider.add_span_processor(BatchSpanProcessor(otlp_exporter))
+            logger.info("Added OTLP span processor for remote tracing collection")
+        except Exception as e:
+            logger.warning(f"Failed to create OTLP exporter: {str(e)}")
+            logger.warning("Remote tracing collection will not be available")
+    else:
+        logger.info("Jaeger/OTLP export is DISABLED - spans will only be processed locally")
+    
+    if AGENT_MORPHEUS_CONSOLE_SPANS_ENABLED:
+        console_exporter = ConsoleSpanExporter()
+        provider.add_span_processor(BatchSpanProcessor(console_exporter))
+        logger.info("Added console span processor for local debugging")
+    
+    # Set as global tracer provider
+    trace.set_tracer_provider(provider)
+    trace_provider = provider
+    
+    logger.info("Tracing system successfully initialized")
+    return provider
+
+def generate_request_id() -> str:
+    """Generate a unique request ID."""
+    return f"req-{uuid.uuid4().hex[:12]}"
+
+def set_request_context(request_id: Optional[str] = None) -> str:
+    """
+    Set the request ID in the context.
+    If no request_id is provided, generates a new one.
+    Returns the request_id.
+    """
+    if request_id is None:
+        request_id = generate_request_id()
+    
+    request_id_var.set(request_id)
+    return request_id
+
+def get_request_id() -> str:
+    """Get the request ID from the context."""
+    return request_id_var.get()
+
+def get_tracer(module_name: str = "default"):
+    """Get a tracer for the specified module."""
+    return trace.get_tracer(module_name)
+
+def record_attribute(category: str, key: str, value: Any) -> None:
+    """
+    Record an attribute on the current span.
+    Prefix attribute name with category for better organization.
+    """
+    if not AGENT_MORPHEUS_TRACING_ENABLED:
+        return
+    
+    current_span = trace.get_current_span()
+    if current_span and current_span.is_recording():
+        # Format attribute name as category.key
+        attribute_name = f"{category}.{key}" if category else key
+        current_span.set_attribute(attribute_name, value)
+
+def record_attributes(attributes: Dict[str, Any]) -> None:
+    """
+    Record multiple attributes on the current span.
+    Input is a dictionary of attribute name-value pairs.
+    """
+    if not AGENT_MORPHEUS_TRACING_ENABLED or not attributes:
+        return
+    
+    current_span = trace.get_current_span()
+    if current_span and current_span.is_recording():
+        for key, value in attributes.items():
+            current_span.set_attribute(key, value)
```

Comment:
@nemerna I Don't see it's being used anywhere...
Why is it still defined in the code ?

Reply:
OK , Please do, 10x.